modules/products/rider/src/main/kotlin/com/metalbear/mirrord/products/rider/RiderPatchCommandLineExtension.kt

```diff
@@ -33,8 +33,13 @@ class RiderPatchCommandLineExtension : PatchCommandLineExtension {
 
         service.execManager.wrapper("rider", commandLine.environment).apply {
             this.wsl = wsl
-        }.start()?.first?.let { env ->
-            for (entry in env.entries.iterator()) {
+        }.start()?.let { executionInfo ->
+            executionInfo.envToUnset?.let { envToUnset ->
+                for (key in envToUnset.iterator()) {
+                    executionInfo.environment.remove(key)
+                }
```

Comment:
This will remove the variable only if it was fetched from the remote

Reply:
We have access to mutable env map here, via `commandLine.environment`. We do the removal right